providers/flagd/pkg/service/in_process/service.go

```diff
@@ -113,6 +114,12 @@ func (i *InProcess) Init() error {
 			case data := <-syncChan:
 				// re-syncs are ignored as we only support single flag sync source
 				changes, _, err := i.evaluator.SetState(data)
+				if data.SyncContext != nil {
+					i.mtx.Lock()
+					i.contextValues = data.SyncContext.AsMap()
```

Comment:
In java we have an optional lambda to manipulate this... I think it should be simple to add it here.